docs/tasks/administer-cluster/dns-custom-nameservers.md

```diff
@@ -142,6 +183,197 @@ data:
+as the logging level and use
+[kubernetes issues](https://github.com/kubernetes/kubernetes/issues)
+to report unexpected errors.
+
+### Is DNS service up?
+
+Verify that the DNS service is up by using the `kubectl get service` command.
+
+```
+kubectl get svc --namespace=kube-system
+NAME          CLUSTER-IP     EXTERNAL-IP   PORT(S)             AGE
+...
+kube-dns      10.0.0.10      <none>        53/UDP,53/TCP        1h
+...
+```
+
+If you have created the service or in the case it should be created by default
+but it does not appear, see 
+[debugging services](/docs/tasks/debug-application-cluster/debug-service/) for
+more information.
+
+### Are DNS endpoints exposed?
+
+You can verify that DNS endpoints are exposed by using the `kubectl get endpoints`
+command.
+
+```
+kubectl get ep kube-dns --namespace=kube-system
+NAME       ENDPOINTS                       AGE
+kube-dns   10.180.3.17:53,10.180.3.17:53    1h
+```
+
+If you do not see the endpoints, see endpoints section in the
+[debugging services](/docs/tasks/debug-application-cluster/debug-service/) documentation.
+
+For additional Kubernetes DNS examples, see the
+[cluster-dns examples](https://github.com/kubernetes/examples/tree/master/staging/cluster-dns)
+in the Kubernetes GitHub repository.
+
+## Known issues
+
+Kubernetes installs do not configure the nodes' resolv.conf files to use the
+cluster DNS by default, because that process is inherently distro-specific.
+This should probably be implemented eventually.
+
+Linux's libc is impossibly stuck ([see this bug from
+2005](https://bugzilla.redhat.com/show_bug.cgi?id=168253)) with limits of just
+3 DNS `nameserver` records and 6 DNS `search` records.  Kubernetes needs to
+consume 1 `nameserver` record and 3 `search` records.  This means that if a
+local installation already uses 3 `nameserver`s or uses more than 3 `search`es,
+some of those settings will be lost.  As a partial workaround, the node can run
+`dnsmasq` which will provide more `nameserver` entries, but not more `search`
+entries.  You can also use kubelet's `--resolv-conf` flag.
+
+If you are using Alpine version 3.3 or earlier as your base image, DNS may not
+work properly owing to a known issue with Alpine.
+Check [here](https://github.com/kubernetes/kubernetes/issues/30215)
+for more information.
+
+## Kubernetes Federation (Multiple Zone support)
+
+Release 1.3 introduced Cluster Federation support for multi-site Kubernetes
+installations. This required some minor (backward-compatible) changes to the
+way the Kubernetes cluster DNS server processes DNS queries, to facilitate
+the lookup of federated services (which span multiple Kubernetes clusters).
+See the [Cluster Federation Administrators' Guide](/docs/concepts/cluster-administration/federation/)
+for more details on Cluster Federation and multi-site support.
+
+## References
+
+- [Docs for the DNS cluster addon](http://releases.k8s.io/{{page.githubbranch}}/cluster/addons/dns/README.md)
+
```

Comment:
Probably worth linking back to the concept doc (dns-pod-service.md)?

Reply:
done.